smithy-build/src/main/java/software/amazon/smithy/build/plugins/BuildInfoPlugin.java

```diff
@@ -77,20 +79,35 @@ private static Node serializeBuildInfo(PluginContext context) {
         return mapper.serialize(info);
     }
 
-    private static List<ShapeId> getTraitShapeIds(Model model) {
+    private static Set<ShapeId> getTraitShapeIdsSet(Model model) {
```

Comment:
I would keep the method as `getTraitShapeIds` and use `result`/`traits` still instead of `traitIdsSet`. The only place where you convert the set to list, you can just do `new ArrayList<>(traits)`